webserver/views/api/v1/test/test_datasets.py

```diff
@@ -79,3 +107,33 @@ def test_create_dataset(self, create_from_dict):
         self.assertEqual(resp.status_code, 200)
         expected = {"success": True, "dataset_id": "6b6b9205-f9c8-4674-92f5-2ae17bcb3cb0"}
         self.assertEqual(resp.json, expected)
+
+    @mock.patch("db.dataset.add_recordings")
+    def test_add_recordings(self, add_recordings):
+        """Successfully add recordings. """
+        id = db.dataset.create_from_dict(self.test_data, author_id=self.test_user_id)
+        ds = db.dataset.get(id)
+        self.temporary_login(self.test_user_id)
+        submit = json.dumps({"class_name": "Class1",
+            "recordings": ["1c085555-3805-428a-982f-e14e0a2b18e6"]
+        })
+        url = '/api/v1/datasets/%s/recordings' %(str(id))
+        resp = self.client.put(url, data=submit, content_type='application/json')
+        self.assertEqual(resp.status_code, 200)
+        expected = {"success": True, "message": "Recording(s) added."}
```

Comment:
Here you are testing that the response from the client is correct, but like your db tests you are not checking the side effects of this call. In python if you are using mocks you can use the `assert_called_once_with` method: https://github.com/vibhorbajpai/acousticbrainz-server/blob/6bd08b330251e0a1cf7a9d06280e84db66370bf6/webserver/views/api/v1/test/test_datasets.py#L96

In the API tests we use mocks so that we don't access the database. This way we can have two separate types of tests: one to check that the database access methods correctly read and write to the database, and one to check that the API correctly calls the database methods. In the API tests we mock the database access methods. This means that you don't need to populate the database here (e.g. with `self.test_data`)